<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>main_tor_util</class>
 <widget class="QWidget" name="main_tor_util">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>420</width>
    <height>460</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>420</width>
    <height>460</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>420</width>
    <height>460</height>
   </size>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>TOR(The Onion Router) Utility</string>
  </property>
  <property name="windowIcon">
   <iconset theme="tor-util">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <widget class="QTabWidget" name="tabwidget_main">
   <property name="geometry">
    <rect>
     <x>5</x>
     <y>50</y>
     <width>410</width>
     <height>355</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>410</width>
     <height>355</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::ClickFocus</enum>
   </property>
   <property name="toolTip">
    <string/>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="tab_send_command">
    <property name="toolTip">
     <string>Send API Commands to TOR Router over network</string>
    </property>
    <attribute name="title">
     <string>Send &amp;Command</string>
    </attribute>
    <widget class="QLineEdit" name="text_host_send">
     <property name="geometry">
      <rect>
       <x>95</x>
       <y>10</y>
       <width>145</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Address of TOR daemon</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNoAutoUppercase</set>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="maxLength">
      <number>64</number>
     </property>
     <property name="cursorPosition">
      <number>0</number>
     </property>
    </widget>
    <widget class="QLineEdit" name="text_port_send">
     <property name="geometry">
      <rect>
       <x>325</x>
       <y>10</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Port of TOR daemon</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly|Qt::ImhPreferNumbers</set>
     </property>
     <property name="inputMask">
      <string notr="true">99999</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="maxLength">
      <number>5</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_host_send">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>55</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Address of TOR daemon</string>
     </property>
     <property name="text">
      <string>Host:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_port_send">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>10</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Port of TOR daemon</string>
     </property>
     <property name="text">
      <string>Port:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_action_send">
     <property name="geometry">
      <rect>
       <x>95</x>
       <y>50</y>
       <width>145</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Select What To Send</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_action_send">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>75</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Select What To Send</string>
     </property>
     <property name="text">
      <string>Action:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="button_send">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>80</y>
       <width>70</width>
       <height>45</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Execute Command</string>
     </property>
     <property name="text">
      <string>&amp;Send</string>
     </property>
     <property name="icon">
      <iconset theme="media-playback-start"/>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="text_output_send">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>130</y>
       <width>395</width>
       <height>185</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="toolTip">
      <string>Return Values From TOR Daemon</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="text_password_send">
     <property name="geometry">
      <rect>
       <x>95</x>
       <y>90</y>
       <width>145</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Optional: TOR API Password. Omitted if Blank</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhHiddenText|Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="maxLength">
      <number>32767</number>
     </property>
     <property name="echoMode">
      <enum>QLineEdit::Password</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_password_send">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>75</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Optional: TOR API Password. Omitted if Blank</string>
     </property>
     <property name="text">
      <string>Password:</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_gen_password_hash">
    <property name="toolTip">
     <string>Generate API Password for torrc</string>
    </property>
    <attribute name="title">
     <string>Generate &amp;Password</string>
    </attribute>
    <widget class="QLineEdit" name="text_password_hash">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>10</y>
       <width>225</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Password to hash</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhHiddenText|Qt::ImhNoAutoUppercase|Qt::ImhNoPredictiveText|Qt::ImhSensitiveData</set>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="maxLength">
      <number>64</number>
     </property>
     <property name="echoMode">
      <enum>QLineEdit::Password</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_password_hash">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Password to hash</string>
     </property>
     <property name="text">
      <string>Password:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="button_gen_password_hash">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>10</y>
       <width>80</width>
       <height>30</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Generate Hash For torrc From Password</string>
     </property>
     <property name="text">
      <string>&amp;Generate</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="text_output_hash">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>50</y>
       <width>395</width>
       <height>265</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="toolTip">
      <string>Generated Hash. Paste This in torrc</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QLabel" name="label_name">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>180</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>App Name</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_version">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>10</y>
     <width>85</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>ver</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>5</x>
     <y>30</y>
     <width>410</width>
     <height>15</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="button_clear">
   <property name="geometry">
    <rect>
     <x>345</x>
     <y>410</y>
     <width>70</width>
     <height>45</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="toolTip">
    <string>Clear Outputs</string>
   </property>
   <property name="text">
    <string>C&amp;lear</string>
   </property>
   <property name="icon">
    <iconset theme="editclear"/>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
  </widget>
  <action name="action_api_send">
   <property name="text">
    <string>_api_send</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_gen_password">
   <property name="text">
    <string>_gen_password</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="action_clear_display">
   <property name="text">
    <string>_clear_display</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="action_exit_cleanup">
   <property name="text">
    <string>_exit_cleanup</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>text_host_send</tabstop>
  <tabstop>text_port_send</tabstop>
  <tabstop>combo_action_send</tabstop>
  <tabstop>text_password_send</tabstop>
  <tabstop>button_send</tabstop>
  <tabstop>text_password_hash</tabstop>
  <tabstop>button_gen_password_hash</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>button_gen_password_hash</sender>
   <signal>clicked()</signal>
   <receiver>action_gen_password</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_send</sender>
   <signal>clicked()</signal>
   <receiver>action_api_send</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_clear</sender>
   <signal>clicked()</signal>
   <receiver>action_clear_display</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>382</x>
     <y>434</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>main_tor_util</sender>
   <signal>destroyed()</signal>
   <receiver>action_exit_cleanup</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
